staking/programs/staking/src/lib.rs

```diff
@@ -721,6 +722,108 @@ pub mod staking {
         Ok(())
     }
 
+    pub fn slash_account(
+        ctx: Context<SlashAccount>,
+        target_with_parameters: TargetWithParameters,
+        // a number between 0 and 1 with 6 decimals of precision
+        // TODO: use fract64 instead of u64
+        slash_ratio: u64,
+    ) -> Result<u64> {
+        let stake_account_positions = &mut ctx.accounts.stake_account_positions.load_mut()?;
+        let next_index = &mut ctx.accounts.stake_account_metadata.next_index;
+
+        let mut total_slashed = 0;
+
+        let mut i: usize = 0;
+        while i < usize::from(*next_index) {
+            msg!("i: {}", i);
+            let position = stake_account_positions.read_position(i)?;
+
+            if let Some(position_data) = position {
+                if position_data.target_with_parameters == target_with_parameters {
+                    let amount = position_data.amount;
+                    // TODO: use constants
+                    let to_slash: u64 =
+                        ((u128::from(amount) * u128::from(slash_ratio) + 1_000_000 - 1)
```

Comment:
I'd round down just for the sake of simplicty

Reply:
Done